app/src/main/java/co/touchlab/droidconandroid/shared/presenter/UserDetailPresenter.java

```diff
@@ -0,0 +1,40 @@
+package co.touchlab.droidconandroid.shared.presenter;
+
+import com.google.j2objc.annotations.Weak;
+
+import co.touchlab.droidconandroid.shared.network.dao.UserAccountInfo;
+import co.touchlab.droidconandroid.shared.tasks.FindUserTask;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.annotations.NonNull;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class UserDetailPresenter {
+
+    @Weak
+    private UserDetailHost host;
+    private FindUserTask task;
+    private CompositeDisposable disposables = new CompositeDisposable();
+
+    public UserDetailPresenter(UserDetailHost host, FindUserTask task) {
+        this.host = host;
+        this.task = task;
+    }
+
+    public void findUser(@NonNull final Long userId) {
+        disposables.add(Observable.zip(task.loadUserAccount(userId),
+                task.loadUserInfo(userId), UserAccountInfo::new)
+                .flatMap(task::saveUserResponse)
```

Comment:
I would consider moving the "saveUserResponse" operation out of this class. If you think about it, caching the new value of the UserAccount into the db isn't part of taking the data and presenting it to the view, which is all the presenters should be concerned with.

You could instead move the save logic up in to the loadUserAccount stream in FindUserTask that way the task is responsible for doing all of the data operations, not the presenter. loadUserAccount() could zip the DB and Network calls, then do the saveUserResponse, then return the final UserAccount. at that point loadUserAccount is the only outward facing method in that task, which make sense because thats all the presenter cares about, "getting a UserAccount object"